### PR TITLE
[Internal] E2EBenchmark: Add reads part of benchmark

### PR DESCRIPTION
**Pull request by kirankumarkolli**
- Add Read workload 
- Latency calculation (Leveraging HdrHistogram NuGet)
- Diagnostics integration into EventSource for high latency operations (available through dotnet trace)
- Pluggable workload
- Interactive stats are for diff payload

Sample output will be like 
image

### CODE REVIEW

File: Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/BenchmarkLatencyEventSource.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+//------------------------------------------------------------
+
+namespace CosmosBenchmark
+{
+    using System;
+    using System.Diagnostics.Tracing;
+    using Microsoft.Azure.Cosmos;
+
+    [EventSource(Name = "Azure.Cosmos.Benchmark")]
+    internal class BenchmarkLatencyEventSource : EventSource
```
Comment (j82w) [inline Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/BenchmarkLatencyEventSource.cs:12]: Should the current benchmarks be checked in when doing an operation on a specific Azure VM?

File: Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/BenchmarkConfig.cs [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Runtime;
+    using CommandLine;
+
+    public class BenchmarkConfig
+    {
+        [Option('w', Required = true, HelpText = "Workload type insert, read")]
```
[the remark this answers is not in the document]
Comment (kirankumarkolli) [reply Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/BenchmarkConfig.cs:15]: Workload type.

File: Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/BenchmarkLatencyEventSource.cs (hunk 1 above)
Comment (kirankumarkolli) [reply Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/BenchmarkLatencyEventSource.cs:12]: Not sure understood the question.

Is the question is to trace VM details in output?
Its a nice to have, will add separate PR.

File: Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/InsertBenchmarkOperation.cs [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+namespace CosmosBenchmark
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Cosmos;
+
+    internal class InsertBenchmarkOperation : IBenchmarkOperatrion
```
[the remark this answers is not in the document]
Comment (kirankumarkolli) [reply Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/InsertBenchmarkOperation.cs:13]: +1, future work

**Pull request merged**